components/form.tsx

```diff
@@ -2,14 +2,90 @@
 
 import Link from 'next/link';
 import Image from 'next/image';
+import { useForm } from 'react-hook-form';
+import type { SubmitHandler } from 'react-hook-form';
+import { z } from 'zod';
+import { zodResolver } from '@hookform/resolvers/zod';
 
-import { Input, Button } from '@/components';
+import { Input, Button, Spinner, Success } from '@/components';
+
+interface FormFields {
+  email: string;
+  username: string;
+  password: string;
+  confirmPassword: string;
+}
+
+export const UserSchema = z
+  .object({
+    email: z
+      .string()
+      .min(1, 'Email is required.')
+      .email('Please enter a valid email address.')
+      .regex(
+        /^[a-zA-Z0-9._%+-]+@(gmail\.com|hotmail\.com|lightit\.io)$/,
+        'Email must be from one of the allowed domains: gmail.com, hotmail.com, or lightit.io.',
+      )
+      .trim(),
```

Comment:
Maybe we can do something like this so its easier if we add more domains in the future

```suggestion
const emailDomains = ['@gmail.com', '@hotmail.com', '@lightit.io'];


export const UserSchema = z
  .object({
    email: z
      .string()
      .min(1, 'Email is required.')
      .email('Please enter a valid email address.')
      .refine((value) => {
    return emailDomains.some(domain => value.endsWith(domain));
  }, {
    message: 'The email must be valid and have a domain of @gmail.com, @hotmail.com, or @lightit.io',
  })
  .trim(),
```